src/python/pants/backend/python/goals/package_pex_binary.py

```diff
@@ -121,22 +126,34 @@ async def package_pex_binary(
     pex_binary_defaults: PexBinaryDefaults,
     union_membership: UnionMembership,
 ) -> BuiltPackage:
-    resolved_entry_point, transitive_targets = await MultiGet(
-        Get(ResolvedPexEntryPoint, ResolvePexEntryPointRequest(field_set.entry_point)),
-        Get(TransitiveTargets, TransitiveTargetsRequest([field_set.address])),
+    resolved_pex_entry_point_get = Get(
+        ResolvedPexEntryPoint, ResolvePexEntryPointRequest(field_set.entry_point)
     )
 
+    if field_set.include_sources.value:
+        resolved_entry_point, transitive_targets = await MultiGet(
+            resolved_pex_entry_point_get,
+            Get(TransitiveTargets, TransitiveTargetsRequest([field_set.address])),
+        )
+        dep_targets = tuple(transitive_targets.dependencies)
+        depends = "transitively depends"
+    else:
+        resolved_entry_point, dependencies_targets = await MultiGet(
```

Comment:
If `include_sources=False` why are we checking at all? Would we pull in a files target that is a direct dep of a `pex_binary` in that case?

Reply:
K. simplified this. Thanks for asking. This is much cleaner now.